### PR TITLE
Add Prayer indicator plugin by merging prayer flick plugin

### LINKED ISSUE #1537: Show active prayers as infoboxes
As requested on discord, show an infobox for each active prayer (konduit does this for example).

Should be fairly easy since everything is already available (fetching sprites, getting active prayer).

### PR DESCRIPTION
**Pull request by raqes**
## Usage
This will add a Prayer Indicator plugin to show the current prayers being used by the player in infoboxes.
Also will this merge with the prayer flicking plugin.
The player can opt out to not show overheads in the infoboxes.

## Related

Fix #1537

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+@PluginDescriptor(
+	name = "Prayer"
+)
+public class PrayerPlugin extends Plugin
+{
+	private PrayerCounter[] prayerCounter = new PrayerCounter[PrayerType.values().length];
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java]: Mark this final

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	@Override
+	protected void startUp()
+	{
+		for (PrayerType prayerType : PrayerType.values())
+		{
+			prayerCounter[prayerType.ordinal()] = new PrayerCounter(this, prayerType);
+			spriteManager.getSpriteAsync(prayerType.getSpriteID(), 0,
+				prayerCounter[prayerType.ordinal()].imageConsumer);
+		}
+	}
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java]: Reset everything on shutDown

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+		if (config.prayerIndicator())
+		{
+			removeIndicators();
+
+			for (Prayer prayer : Prayer.values())
+			{
+				if (client.isPrayerActive(prayer))
+				{
+					PrayerType prayerType = PrayerType.getPrayerType(prayer);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java]: mark this final

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerType.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+	private String name;
+	private Prayer prayer;
+	private String description;
+	private int spriteID;
+	private boolean overhead;
+
+	public static PrayerType getPrayerType(Prayer prayer)
+	{
+		for  (PrayerType prayerType : values())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerType.java]: Cache this to static map

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerType.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+		{
+			if (prayerType.prayer == prayer)
+			{
+				return prayerType;
+			}
+		}
+		return null;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerType.java]: Add empty line above

**Comment by deathbeam** [comment]
You should probably split the renaming of the plugin and addition of new features to the plugin to 2 separate commits

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java [hunk 6]
```diff
@@ -22,32 +22,44 @@
...
+public class PrayerCounter extends Counter
 {
-	@Inject
-	private PrayerFlickOverlay overlay;
+	@Getter
+	private PrayerType prayerType;
+	private BufferedImage image;
+	private static String text = "";
+
+	Consumer<BufferedImage> imageConsumer = (image) -> this.image = image;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: Why is this here? just create setter for image and reference that method.

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java [hunk 7]
```diff
@@ -22,32 +22,44 @@
...
-)
-public class PrayerFlickPlugin extends Plugin
+public class PrayerCounter extends Counter
 {
-	@Inject
-	private PrayerFlickOverlay overlay;
+	@Getter
+	private PrayerType prayerType;
+	private BufferedImage image;
+	private static String text = "";
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: Why is this static? or even present here?

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java [hunk 8]
```diff
@@ -22,32 +22,44 @@
...
-@PluginDescriptor(
-	name = "Prayer Flicking"
-)
-public class PrayerFlickPlugin extends Plugin
+public class PrayerCounter extends Counter
 {
-	@Inject
-	private PrayerFlickOverlay overlay;
+	@Getter
+	private PrayerType prayerType;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: Mark this final

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java (hunk 6 above)
Comment (raqes) [reply runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: To get the image async from the cache. Eg. at loading up the game it won't retrieve the image.

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: Yes. `PrayerCounter::setImage` == `Consumer<BufferedImage>`

Comment (raqes) [reply runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: You learn something new everday, thanks ;)

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+import java.awt.image.BufferedImage;
+import lombok.Getter;
+import lombok.Setter;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.ui.overlay.infobox.Counter;
+
+public class PrayerCounter extends Counter
+{
+	@Getter
+	final private PrayerType prayerType;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: private final, or at least that is the format we use everywhere

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+	private BufferedImage image;
+
+	PrayerCounter(Plugin plugin, PrayerType prayerType)
+	{
+		super(null, plugin, "");
+		this.prayerType = prayerType;
+	}
+
+	@Override
+	public String toString()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: is toString really necessary?

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java [hunk 11]
```diff
@@ -25,29 +26,127 @@
...
 import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
 
 @PluginDescriptor(
-	name = "Prayer Flicking"
+	name = "Prayer"
 )
 public class PrayerPlugin extends Plugin
 {
+	final private PrayerCounter[] prayerCounter = new PrayerCounter[PrayerType.values().length];
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java]: private final

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java [hunk 12]
```diff
@@ -25,29 +26,127 @@
...
+
+				if (prayerCounter[ord] == null)
+				{
+					PrayerCounter counter = prayerCounter[ord] = new PrayerCounter(this, prayerType);
+					spriteManager.getSpriteAsync(prayerType.getSpriteID(), 0,
+						counter::setImage);
+					infoBoxManager.addInfoBox(counter);
+				}
+			}
+			else
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerPlugin.java]: else if { instead of else { if {

File: runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java (hunk 10 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java]: Ah, i would probably use lombok then
